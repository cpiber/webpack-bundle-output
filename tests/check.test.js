const { webpack, file } = require('./shared');
const { getFixturePathSync } = require('jest-fixtures');

test('single file', async () => {
  const { map } = await webpack({
    entry: getFixturePathSync(__dirname, 'basic'),
  });
  expect(file(map)).toMatchSnapshot();
});

test('includes', async () => {
  const { map } = await webpack({
    entry: getFixturePathSync(__dirname, 'includes'),
  });
  expect(file(map)).toMatchSnapshot();
});

test('import', async () => {
  const { map } = await webpack({
    entry: getFixturePathSync(__dirname, 'import'),
  });
  expect(file(map)).toMatchSnapshot();
});

test('multiple entries basic', async () => {
  const { map } = await webpack({
    entry: {
      basic: getFixturePathSync(__dirname, 'basic'),
      includes: getFixturePathSync(__dirname, 'includes'),
    },
  });
  expect(file(map)).toMatchSnapshot();
});

test('multiple entries cross-files', async () => {
  const { map } = await webpack({
    entry: {
      basic: getFixturePathSync(__dirname, 'basic'),
      includes: getFixturePathSync(__dirname, 'includes'),
      cross: getFixturePathSync(__dirname, 'cross'),
    },
  });
  expect(file(map)).toMatchSnapshot();
});

test('lib', async () => {
  const { map } = await webpack({
    entry: getFixturePathSync(__dirname, 'lib'),
  });
  expect(file(map)).toMatchSnapshot();
});

test('loader', async () => {
  const { map } = await webpack({
    entry: getFixturePathSync(__dirname, 'loader'),
  });
  expect(file(map)).toMatchSnapshot();
});

test('ts', async () => {
  const { map } = await webpack({
    entry: getFixturePathSync(__dirname, 'ts'),
    resolve: {
      extensions: [".ts", ".tsx", ".js"],
    },
    module: {
      rules: [
        { test: /\.tsx?$/, loader: "ts-loader", options: { configFile: getFixturePathSync(__dirname, 'ts/tsconfig.json') } },
      ],
    },
  });
  expect(file(map)).toMatchSnapshot();
});

test('ts loader', async () => {
  const { map } = await webpack({
    entry: getFixturePathSync(__dirname, 'loader-ts'),
    resolve: {
      extensions: [".ts", ".tsx", ".js"],
    },
    module: {
      rules: [
        { test: /\.tsx?$/, loader: "ts-loader", options: { configFile: getFixturePathSync(__dirname, 'ts/tsconfig.json') } },
      ],
    },
  });
  expect(file(map)).toMatchSnapshot();
});

test('async', async () => {
  const { map } = await webpack({
    entry: getFixturePathSync(__dirname, 'async'),
  });
  expect(file(map)).toMatchSnapshot();
});

test('async loader', async () => {
  const { map } = await webpack({
    entry: getFixturePathSync(__dirname, 'loader-async'),
  });
  expect(file(map)).toMatchSnapshot();
});