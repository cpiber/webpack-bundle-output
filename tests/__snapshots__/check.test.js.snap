// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async 1`] = `
{
  "tests/__fixtures__/async/index.js": [
    "tests/out/5b71205016f3f4d021c84fcaa0c3882f/main.js",
  ],
  "tests/__fixtures__/basic/index.js": [
    "tests/out/5b71205016f3f4d021c84fcaa0c3882f/tests___fixtures___basic_index_js.js",
  ],
}
`;

exports[`async loader 1`] = `
{
  "tests/__fixtures__/basic/index.js": [
    "tests/out/124742d27df8c35bd28c3f5685f945ba/node_modules_pnpm_raw-loader_4_0_2_webpack_5_98_0_node_modules_raw-loader_dist_cjs_js_tests__-3d3f28.js",
  ],
  "tests/__fixtures__/loader-async/index.js": [
    "tests/out/124742d27df8c35bd28c3f5685f945ba/main.js",
  ],
}
`;

exports[`import 1`] = `
{
  "tests/__fixtures__/import/index.js": [
    "tests/out/74b4d3d0f5515a688112d490508c0f0d/main.js",
  ],
  "tests/__fixtures__/import/other.js": [
    "tests/out/74b4d3d0f5515a688112d490508c0f0d/main.js",
  ],
}
`;

exports[`includes 1`] = `
{
  "tests/__fixtures__/includes/index.js": [
    "tests/out/1913f09eb081a8f3fded828c8294387a/main.js",
  ],
  "tests/__fixtures__/includes/other.js": [
    "tests/out/1913f09eb081a8f3fded828c8294387a/main.js",
  ],
}
`;

exports[`lib 1`] = `
{
  "node_modules/.pnpm/jquery@3.7.1/node_modules/jquery/dist/jquery.js": [
    "tests/out/144bfedf5c434a6a9e222b67d1fddd4f/main.js",
  ],
  "tests/__fixtures__/lib/index.js": [
    "tests/out/144bfedf5c434a6a9e222b67d1fddd4f/main.js",
  ],
}
`;

exports[`loader 1`] = `
{
  "tests/__fixtures__/basic/index.js": [
    "tests/out/99fd1a68ed5b12e4a771c180bc6998cc/main.js",
  ],
  "tests/__fixtures__/loader/index.js": [
    "tests/out/99fd1a68ed5b12e4a771c180bc6998cc/main.js",
  ],
}
`;

exports[`multiple entries basic 1`] = `
{
  "tests/__fixtures__/basic/index.js": [
    "tests/out/e6de0cf1a998c217e85729459c80900f/basic.js",
  ],
  "tests/__fixtures__/includes/index.js": [
    "tests/out/e6de0cf1a998c217e85729459c80900f/includes.js",
  ],
  "tests/__fixtures__/includes/other.js": [
    "tests/out/e6de0cf1a998c217e85729459c80900f/includes.js",
  ],
}
`;

exports[`multiple entries cross-files 1`] = `
{
  "tests/__fixtures__/basic/index.js": [
    "tests/out/fa1abf78780b2d63a098f4960aa06ee1/basic.js",
    "tests/out/fa1abf78780b2d63a098f4960aa06ee1/cross.js",
  ],
  "tests/__fixtures__/cross/index.js": [
    "tests/out/fa1abf78780b2d63a098f4960aa06ee1/cross.js",
  ],
  "tests/__fixtures__/includes/index.js": [
    "tests/out/fa1abf78780b2d63a098f4960aa06ee1/includes.js",
    "tests/out/fa1abf78780b2d63a098f4960aa06ee1/cross.js",
  ],
  "tests/__fixtures__/includes/other.js": [
    "tests/out/fa1abf78780b2d63a098f4960aa06ee1/includes.js",
    "tests/out/fa1abf78780b2d63a098f4960aa06ee1/cross.js",
  ],
}
`;

exports[`single file 1`] = `
{
  "tests/__fixtures__/basic/index.js": [
    "tests/out/817377307c84c12762cca0b37318c635/main.js",
  ],
}
`;

exports[`ts 1`] = `
{
  "tests/__fixtures__/ts/index.ts": [
    "tests/out/b205b5d61358817da8761615bae42929/main.js",
  ],
}
`;

exports[`ts loader 1`] = `
{
  "tests/__fixtures__/loader-ts/index.js": [
    "tests/out/8a2590ef8037e73cdcf5209cd3628c84/main.js",
  ],
  "tests/__fixtures__/ts/index.ts": [
    "tests/out/8a2590ef8037e73cdcf5209cd3628c84/main.js",
  ],
  "tests/__fixtures__/ts/tsconfig.json": [
    "tests/out/8a2590ef8037e73cdcf5209cd3628c84/main.js",
  ],
}
`;
